components/Alert/AlertsAPI.tsx

```diff
@@ -0,0 +1,234 @@
+    alerts,
+    item => item.key as string,
+    transitionConfigs
+  )
+
+  // Return container based on placement prop
+  const getWrapper = () => {
+    switch (props.placement) {
+      case "topLeft":
+        return TopLeftBox
+      case "topRight":
+        return TopRightBox
+      case "bottomLeft":
+        return BottomLeftBox
+      case "bottomRight":
+        return BottomRightBox
+      default:
+        return BottomLeftBox
+    }
+  }
+
+  const Wrapper = getWrapper()
+
+  return (
+    <Wrapper>
+      {transitionProps.map(({ key, item, props: { life, ...style } }: any) => {
+        let onClose = item.onClose
+        const curOnClose = onClose
+        onClose = (e?: Event): void => {
+          if (curOnClose) {
+            curOnClose(e)
+          }
+          if (cancelMap.has(item)) {
+            cancelMap.get(item)()
+          }
+        }
+        item.onClose = onClose
+
+        return (
+          <AnimatedDiv key={key} style={style}>
+            <Div ref={ref => ref && refMap.set(item, ref)}>
+              <Alert {...item} />
+            </Div>
+          </AnimatedDiv>
+        )
+      })}
+    </Wrapper>
+  )
+}
+
+// Wrapper Component for particular corner, Using React Class Component to access methods using API (callback ref).
+class AlertsWrapper extends React.Component<
+  AlertsWrapperProps,
+  AlertsWrapperStates
+> {
+  placement: placementType
+
+  constructor(props: AlertsWrapperProps) {
+    super(props)
+    this.state = {
+      alerts: [],
+    }
+    this.placement = this.props.placement ? this.props.placement : "bottomLeft"
+    this.add.bind(this)
+    this.remove.bind(this)
+    this.add.bind(this)
+  }
+
+  // Passed to API, add Alert to state and generate unique key
+  add = (alertProps: AlertProps): void => {
+    const key: string = generateKey()
+    alertProps = { key: key, ...alertProps }
+
+    // Closing Alert will call remove method of this class to update state
+    const curOnClose = alertProps.onClose
+    alertProps.onClose = event => {
+      this.remove(key)
+      if (curOnClose) {
+        curOnClose(event)
+      }
+    }
+
+    const newState = [alertProps, ...this.state.alerts]
+
+    this.setState({
+      alerts: newState,
+    })
+  }
+
+  // Passed to API, remove Element from state using key
+  remove = (key: string): void => {
+    const newState = this.state.alerts.filter(i => i.key !== key)
+    this.setState({ alerts: newState })
+  }
+
+  render() {
+    return (
+      <TransitionWrapper
+        placement={this.placement}
+        alerts={this.state.alerts}
+      />
+    )
+  }
+}
+
+// Alerts save `AlertsWrapper` of 4 corners using [placement]: AlertsWrapperInstance
+const alertsWrappers: { [placement: string]: AlertAPIInterface } = {}
+// Div to refer new div element in below open function
+let div: HTMLElement
+
+// Open function for the API
+function open(alertProps: AlertProps) {
+  const placement: string = alertProps.placement as string
```

Comment:
```suggestion
  const { placement } = alertProps
```

Reply:
`alertProps.placement` is of `placementType`, to use it as object key need needed to change in string,
still can be short to `const placement = alertProps.placement as string`